pkg/operator/e2e/main_test.go

```diff
@@ -407,23 +415,42 @@ func testCollectorDeployed(ctx context.Context, t *testContext) {
 		if ds.Status.DesiredNumberScheduled == 0 {
 			return false, nil
 		}
-		var projMatch, clusterMatch, locMatch, collectorArgs bool
+		if ds.Status.NumberReady != ds.Status.DesiredNumberScheduled {
+			return false, nil
+		}
 		for _, c := range ds.Spec.Template.Spec.Containers {
 			if c.Name != "prometheus" {
 				continue
 			}
-			for _, a := range c.Args {
-				if a == "--export.label.project-id=test-proj" {
-					projMatch = true
-				} else if a == "--export.label.cluster=test-cluster" {
-					clusterMatch = true
-				} else if a == "--export.label.location=test-loc" {
-					locMatch = true
-				}
+			// We're mainly interested in the dynamic flags but checking the entire set including
+			// the static ones is ultimately simpler.
+			wantArgs := []string{
+				"--config.file=/prometheus/config_out/config.yaml",
+				"--storage.tsdb.path=/prometheus/data",
+				"--storage.tsdb.no-lockfile",
+				"--storage.tsdb.retention.time=30m",
+				"--storage.tsdb.wal-compression",
+				"--storage.tsdb.min-block-duration=10m",
+				"--storage.tsdb.max-block-duration=10m",
+				fmt.Sprintf("--web.listen-address=:%d", t.collectorPort),
+				"--web.enable-lifecycle",
+				"--web.route-prefix=/",
+				fmt.Sprintf("--export.label.project-id=%s", projectID),
+				fmt.Sprintf("--export.label.cluster=%s", cluster),
```

Comment:
What about `fmt.Sprintf("--export.label.location=%s", location)`?

Reply:
Oh, I see we did add a flag – I did run directly on my machine to verify GCM.
Will take care of it.